src/quickstart.ts

```diff
@@ -1,149 +1,163 @@
+  if (await fs.exists(sourceCodeFilePath)) {
+    throw new Error('Library source file is already present.');
   }
 
-  private async createCMakeListFile() {
-    const init = [
-      'cmake_minimum_required(VERSION 3.0.0)',
-      `project(${this.projectName} VERSION 0.1.0)`,
-      '',
-      'include(CTest)',
-      'enable_testing()',
-      '',
-      this.type == ProjectType.Library ? `add_library(${this.projectName} ${this.projectName}.cpp)`
-                                       : `add_executable(${this.projectName} main.cpp)`,
-      '',
-      'set(CPACK_PROJECT_NAME ${PROJECT_NAME})',
-      'set(CPACK_PROJECT_VERSION ${PROJECT_VERSION})',
-      'include(CPack)',
-      '',
-    ].join('\n');
-
-    return fs.writeFile(this.cmakeFilePath, init);
-  }
+  await fs.writeFile(sourceCodeFilePath, [
+    '#include <iostream>',
+    '',
+    'void say_hello(){',
+    `    std::cout << "Hello, from ${projectInformation.name}!\\n";`,
+    '}',
+    '',
+  ].join('\n'));
+  return sourceCodeFilePath;
+}
 
-  private async createLibrarySourceFile() {
-    return fs.writeFile(this.sourceCodeFilePath, [
-      '#include <iostream>',
-      '',
-      'void say_hello(){',
-      `    std::cout << "Hello, from ${this.projectName}!\\n";`,
-      '}',
-      '',
-    ].join('\n'));
+async function createMainSourceCodeFile(projectInformation: CreateProjectInformation) {
+  const sourceCodeFilePath = path.join(projectInformation.sourceDirectory, 'main.cpp');
+  if (await fs.exists(sourceCodeFilePath)) {
+    throw new Error('Main source file is already present.');
   }
 
-  private async createMainSourceCodeFile() {
-    return fs.writeFile(this.sourceCodeFilePath, [
-      '#include <iostream>',
-      '',
-      'int main(int, char**) {',
-      '   std::cout << "Hello, world!\\n";',
-      '}',
-      '',
-    ].join('\n'));
-  }
+  await fs.writeFile(sourceCodeFilePath, [
+    '#include <iostream>',
+    '',
+    'int main(int, char**) {',
+    '   std::cout << "Hello, world!\\n";',
+    '}',
+    '',
+  ].join('\n'));
+
+  return sourceCodeFilePath;
+}
 
-  private async createExampleSourcecodeFile() {
-    if (!(await fs.exists(this.sourceCodeFilePath))) {
-      switch (this.type) {
-      case ProjectType.Library:
-        return this.createLibrarySourceFile();
-      case ProjectType.Exectable:
-        return this.createMainSourceCodeFile();
-      }
+export async function createProject(projectInformation: CreateProjectInformation):
+    Promise<GeneratedProjectFiles> {
+  const cmakeFile = await createCMakeListFile(projectInformation);
+
+  let sourceFile: string = '';
+  try {
+    if (projectInformation.type == ProjectType.Exectable) {
+      sourceFile = await createMainSourceCodeFile(projectInformation);
+    } else if (projectInformation.type == ProjectType.Library) {
+      sourceFile = await createLibrarySourceFile(projectInformation);
     }
-  }
+  } catch (err) { log.debug(err); }
```

Comment:
Be more careful with error handling here. Instead of catching and throwing for errors from the `createLibrarySourceFile` functions, have them return a union type that contains the error information:

```typescript 
enum CreateFileError {
  FileExists = 1,
}

async function createMainSourceCodeFile(info): Promise<CreateFileError|string> {
  // ... 
}
```

Usage like this:

```typescript 
const result = await createMainSourceCodeFile(projectInformation);
if (result == CreateFileError.FileExists) {
  return result;
}
// It succeeded, so we have a string:
sourceFile = result;
```